plugin/goto_diagnostic.py

```diff
@@ -77,18 +80,61 @@ def input_description(self) -> str:
         return "Goto Diagnostic"
 
 
-class DiagnosticUriInputHandler(sublime_plugin.ListInputHandler):
+ListItemsReturn = Union[List[str], Tuple[List[str], int], List[Tuple[str, Any]], Tuple[List[Tuple[str, Any]], int],
+                        List[sublime.ListInputItem], Tuple[List[sublime.ListInputItem], int]]
+
+
+class PreselectedListInputHandler(sublime_plugin.ListInputHandler, metaclass=ABCMeta):
+    """
+    Similar to ListInputHandler, but allows to preselect a value like some of the input overlays in Sublime Merge.
+    Inspired by https://github.com/sublimehq/sublime_text/issues/5507.
+
+    Subclasses of PreselectedListInputHandler must not implement the `list_items` method, but instead `_list_items`,
+    i.e. just prepend an underscore to the regular `list_items`.
+
+    When an instance of PreselectedListInputHandler is created, it must be given the window as an argument.
+    An optional second argument `initial_value` can be provided to preselect a value.
+    """
+
+    _window = None  # type: Optional[sublime.Window]
+    _initial_value = None  # type: Optional[Union[str, sublime.ListInputItem]]
+    _preselect = False
+
+    def __init__(
+        self, window: sublime.Window, initial_value: Optional[Union[str, sublime.ListInputItem]] = None
+    ) -> None:
+        super().__init__()
+        if initial_value is not None:
+            self._window = window
+            self._initial_value = initial_value
+            self._preselect = True
+
+    def list_items(self) -> ListItemsReturn:
+        if self._preselect and self._initial_value is not None and self._window is not None:
+            self._preselect = False
+            sublime.set_timeout(functools.partial(self._window.run_command, 'select'))
+            return [self._initial_value], 0  # pyright: ignore[reportGeneralTypeIssues]
+        else:
+            return self._list_items()
+
+    @abstractmethod
+    def _list_items(self) -> ListItemsReturn:
```

Comment:
Rather than having two "list_items", I think I would prefer calling it `get_list_items`. And whether there should be a leading underscore or not, I'm not sure so I'm fine with either.